core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -835,91 +836,91 @@ object Parser extends ParserInstances {
 
   /** Repeat this parser 0 or more times
     * note: this can wind up parsing nothing
     */
-  def rep[A](p1: Parser1[A]): Parser[List[A]] =
-    repAs[A, List[A]](p1)
+  def rep0[A](p1: Parser[A]): Parser0[List[A]] =
+    repAs0[A, List[A]](p1)
 
   /** Repeat this parser 0 or more times
     * note: this can wind up parsing nothing
     */
-  def repAs[A, B](p1: Parser1[A])(implicit acc: Accumulator[A, B]): Parser[B] =
-    Impl.Rep(p1, acc)
+  def repAs0[A, B](p1: Parser[A])(implicit acc: Accumulator0[A, B]): Parser0[B] =
+    Impl.Rep0(p1, acc)
 
   /** Repeat this parser 1 or more times
     */
-  def rep1[A](p1: Parser1[A], min: Int): Parser1[NonEmptyList[A]] =
-    repAs1[A, NonEmptyList[A]](p1, min)
+  def rep[A](p1: Parser[A], min: Int): Parser[NonEmptyList[A]] =
+    repAs[A, NonEmptyList[A]](p1, min)
 
   /** Repeat this parser 1 or more times
     */
-  def repAs1[A, B](p1: Parser1[A], min: Int)(implicit acc: Accumulator1[A, B]): Parser1[B] =
-    Impl.Rep1(p1, min, acc)
+  def repAs[A, B](p1: Parser[A], min: Int)(implicit acc: Accumulator[A, B]): Parser[B] =
+    Impl.Rep(p1, min, acc)
 
   /** Repeat 1 or more times with a separator
     */
-  def rep1Sep[A](p1: Parser1[A], min: Int, sep: Parser[Any]): Parser1[NonEmptyList[A]] = {
+  def repSep[A](p1: Parser[A], min: Int, sep: Parser0[Any]): Parser[NonEmptyList[A]] = {
     if (min <= 0) throw new IllegalArgumentException(s"require min > 0, found: $min")
 
-    val rest = (sep.void.with1.soft *> p1).rep(min - 1)
+    val rest = (sep.void.with1.soft *> p1).rep0(min - 1)
     (p1 ~ rest).map { case (h, t) => NonEmptyList(h, t) }
   }
 
   /** Repeat 0 or more times with a separator
     */
-  def repSep[A](p1: Parser1[A], min: Int, sep: Parser[Any]): Parser[List[A]] = {
-    if (min <= 0) rep1Sep(p1, 1, sep).?.map {
+  def rep0Sep[A](p1: Parser[A], min: Int, sep: Parser0[Any]): Parser0[List[A]] = {
+    if (min <= 0) repSep(p1, 1, sep).?.map {
       case None => Nil
       case Some(nel) => nel.toList
     }
-    else rep1Sep(p1, min, sep).map(_.toList)
+    else repSep(p1, min, sep).map(_.toList)
   }
 
   /** parse first then second
     */
-  def product[A, B](first: Parser[A], second: Parser[B]): Parser[(A, B)] =
+  def product0[A, B](first: Parser0[A], second: Parser0[B]): Parser0[(A, B)] =
     first match {
-      case f1: Parser1[A] => product10(f1, second)
+      case f1: Parser[A] => product(f1, second)
       case _ =>
         second match {
-          case s1: Parser1[B] =>
+          case s1: Parser[B] =>
             product01(first, s1)
-          case _ => Impl.Prod(first, second)
+          case _ => Impl.Prod0(first, second)
         }
     }
 
-  /** product with the first argument being a Parser1
+  /** product with the first argument being a Parser
     */
-  def product10[A, B](first: Parser1[A], second: Parser[B]): Parser1[(A, B)] =
-    Impl.Prod1(first, second)
+  def product[A, B](first: Parser[A], second: Parser0[B]): Parser[(A, B)] =
```

Comment:
can we just keep the name product10 here, I think methods like this with several variants it is fine to still use the 1 to make it clearer.